### PR TITLE
Hot reload + Metrics minor improvements

### PR DESCRIPTION
**Pull request by buixor**
Support for reload + systemd reload :
 - shutdown all go-routines
 - serialize buckets state
 - reload all configs
 - restore bucket internal state
 - restart processing routines

Make `cscli metrics` always be displayed in the same order

Add prometheus metric on current bucket count for each scenario.

### CODE REVIEW

File: pkg/csconfig/config.go [hunk 1]
```diff
@@ -55,6 +56,7 @@ func NewCrowdSecConfig() *CrowdSec {
 		LogMode:       "stdout",
 		SQLiteFile:    "./test.db",
```
[the remark this answers is not in the document]
Comment (buixor) [reply pkg/csconfig/config.go:56]: You want the default to be to `/var/lib/...` ?

[the remark this answers is not in the document]
Comment (AlteredCoder) [reply pkg/csconfig/config.go:56]: yes, no ?

File: cmd/crowdsec/serve.go [hunk 2]
```diff
@@ -1,78 +1,131 @@
...
+	if err := os.Remove(tmpFile); err != nil {
+		log.Warningf("Failed to delete temp file (%s) : %s", tmpFile, err)
+	}
 	return nil
 }
 
-func termHandler(sig os.Signal) error {
-	log.Warningf("Shutting down routines")
+func ShutdownRoutines() error {
+	var reterr error
```
Comment (AlteredCoder) [inline cmd/crowdsec/serve.go:91]: maybe we want an array of `error` no ?

if multiple Tomb failed to Kill, the error that will be printed will concern only the last failing one

Comment (buixor) [reply cmd/crowdsec/serve.go:91]: I think single error is better, but I'm going to print the errors on the fly :)

**Pull request merged**